lib/eslint.js

```diff
@@ -807,20 +811,18 @@ module.exports = (function() {
              */
             controller.traverse(ast, {
                 enter: function(node, parent) {
-
                     var comments = api.getComments(node);
 
                     emitCommentsEnter(comments.leading);
                     node.parent = parent;
-                    api.emit(node.type, node);
+                    codePathAnalyzer.enterNode(node);
```

Comment:
OK. There are some note:

- Code path events are fired only when it's changed from the previous one.
- An ast node has several code path segments each AST node in `finally` blocks. So it's possible to fire several `onCodePathSegment` events on a AST node.
- There are both events which are fired before and after the normal AST event.

So I guess the return value is such as:

```js
{
    "before": [
        {"type": "onCodePath:exit", "args": [codePath1]},
        {"type": "onCodePath", "args": [codePath2]},
        {"type": "onCodePathSegment", "args": [segment1, codePath2]},
        {"type": "onCodePathSegment", "args": [segment2, codePath2]}
        // ...
    ],
    "after": [] // similar to `before`
}
```